src/ipc/ipc-common/IPCProxyBase.h

```diff
@@ -43,23 +42,132 @@
 
 namespace facelift {
 
-template<typename AdapterType>
-class IPCProxyBase : public AdapterType, protected IPCProxyBaseBase
+class Monitor;
+class IObserver
```

Comment:
What does this code enable, which the code from the very first commit of that PR did not ?
This commit looks horribly complex. Why not simply moving the code from the first commit into a helper class ?

Reply:
Please don't worry, I take account all comments/notes.
All code will be rewritten.